apps/web/pages/api/social/og/image.tsx

```diff
@@ -21,28 +21,50 @@ export const config = {
   runtime: "edge",
 };
 
+const ZLocale = z.union([z.string(), z.null(), z.undefined()]).transform((val) => val ?? "en");
+
 const meetingSchema = z.object({
   imageType: z.literal("meeting"),
   title: z.string(),
   names: z.string().array(),
   usernames: z.string().array(),
   meetingProfileName: z.string(),
   meetingImage: z.string().nullable().optional(),
+  locale: ZLocale,
 });
 
 const appSchema = z.object({
   imageType: z.literal("app"),
   name: z.string(),
   description: z.string(),
   slug: z.string(),
+  locale: ZLocale,
 });
 
 const genericSchema = z.object({
   imageType: z.literal("generic"),
   title: z.string(),
   description: z.string(),
+  locale: ZLocale,
 });
 
+async function getTranslation({ locale, params }: { locale: string; params: string[] }) {
+  const defaultFn = (value: string) => value;
+  const allLowerCaseNumbersAndUnderscores = /^[a-z0-9_]+$/;
+  const requiresTranslation = params.some((arg) => allLowerCaseNumbersAndUnderscores.test(arg));
+  if (!requiresTranslation) return { t: defaultFn };
+
+  try {
+    const translation = await fetch(`https://cal.com/static/locales/${locale}/common.json`, {});
+    const json = await translation.json();
+    const t = (value: string): string => json[value] ?? value;
```

Comment:
This will try to get the value from the translation but if no value is found it will return the string as is, so this will not only be a safe fallback but if it ever matches a "translated" string it will still return it.